ipatests/test_integration/test_sssd.py

```diff
@@ -403,3 +407,173 @@ def test_trustdomain_disable_disables_subdomain(self):
                 assert 'no such user' in res.stderr_text
         # verify the user can be retrieved after re-enabling trustdomain
         self.master.run_command(['id', user])
+
+
+class TestNestedMembers(IntegrationTest):
+    num_clients = 1
+
+    @classmethod
+    def install(cls, mh):
+        tasks.install_master(cls.master)
+        tasks.install_client(cls.master, cls.clients[0])
+
+    def test_nested_group_members(self):
+        """Nested group memberships should be honoured
+
+        "child2" should be a child of "child1"
+        so that parent child relationship is as follows:
+        "parent"->"child1"->"child2"
+
+        testuser001 is direct member of "parent" and as a result
+        member of "child1" and "child2"".
+        Now if one adds a direct membership to "child1"
+        nothing will change.
+
+        Now if one removes the direct membership to "child1"
+        nothing should change, the memberships should be honored
+        Linked Issue: https://pagure.io/SSSD/sssd/issue/3636
+        """
+        master = self.master
+        client = self.clients[0]
+
+        username = "testuser001"
+        userpasswd = 'Secret123'
+
+        # tmpdir for creating ssh keys for user
+        # since ssh does not work well paramiko
+        tmpdir = tempfile.mkdtemp(prefix="tmp-")
+
+        # add a user and set password
+        tasks.create_active_user(master, username, userpasswd)
+        tasks.kinit_admin(master)
+
+        pub_keys = []
+
+        for i in range(40):
```

Comment:
I don't understand this loop. Why does the test add 40 keys?
Wouldn't a single key be enough for the test?

Reply:
removed the loop.